File: Practica 4/Ej2/suma_n_numeros.c
#include <stdio.h>

int suma_n_primeros_nums(int num)
{
    int total = 0;
    for (int i = 1; i <= num; i++)
    {
        total += i;
    }

    return total;
}

int main()
{
    int num;

    printf("Introduce un número entero positivo: ");
    scanf("%d", &num);

    while (num < 0)
    {
        printf("Error: El número debe ser positivo\n");

        printf("Vuelve a introducirlo: ");
        scanf("%d", &num);
    }

    printf("La suma de los %d primeros enteros es: %d", num, suma_n_primeros_nums(num));
}

File: Practica 6/Ej2/suma_n_numeros.py
def main():
    num = int(input("Introduce un número entero positivo: "))

    while num < 0:
        print("Error: El número debe ser positivo")
        num = int(input("Introduce un número entero positivo: "))

    print(f"La suma de los {num} primeros enteros es {suma_n_primeros_nums(num)}")
    
def suma_n_primeros_nums(num):
    total = 0

    for n in range(num + 1):
        total += n

    return total

main()
